accounts-db/src/accounts_hash.rs

```diff
@@ -994,54 +1005,138 @@ impl<'a> AccountsHasher<'a> {
                 }
             })
             .sum::<usize>();
+
+        (working_set, max_inclusive_num_pubkeys)
+    }
+
+    /// Add next item into hash dedup working set
+    fn add_next_item<'b>(
+        next: &mut Option<ItemLocation<'b>>,
+        new: &mut Option<(usize, usize)>,
+        working_set: &mut Vec<(usize, usize)>,
+        sorted_data_by_pubkey: &[&'b [CalculateHashIntermediate]],
+        pubkey_bin: usize,
+        binner: &PubkeyBinCalculator24,
+    ) {
+        // looping to add next item to working set
+        while let Some(ItemLocation {
+            key,
+            division_index: slot_group_index,
+            offset,
+        }) = next
+        {
+            // if `new key` is less than the min key in the working set, remember current item and process it directly (skipping working set insertion).
+            if let Some((current_min_slot_group_index, current_min_offset)) = working_set.last() {
+                let current_min_key = &sorted_data_by_pubkey[*current_min_slot_group_index]
+                    [*current_min_offset]
+                    .pubkey;
+                if *key < current_min_key {
+                    *new = Some((*slot_group_index, *offset));
+                    break;
+                }
+            }
+
+            let found = working_set.binary_search_by(|(slot_group_index, offset)| {
+                let prob = &sorted_data_by_pubkey[*slot_group_index][*offset].pubkey;
+                (*key).cmp(prob)
+            });
+
+            match found {
+                Err(index) => {
+                    // found a new new key, insert into the working_set. This is O(n/2) on
+                    // average. Theoretically, this operation could be expensive and may be further
+                    // optimized in future.
+                    working_set.insert(index, (*slot_group_index, *offset));
+                    break;
+                }
+                Ok(index) => {
+                    let v = &mut working_set[index];
+                    if v.0 > *slot_group_index {
```

Comment:
I think we should have a type in working_set. v.1 > *slot_group_index would also compile but be totally incorrect.
If it was v.slot_group_index > *slot_group_index, the code is more obviously correct by inspection.

Reply:
done.